pallets/confidential/src/lib.rs

```diff
@@ -79,6 +88,36 @@ decl_module! {
 
         fn deposit_event() = default;
 
+        #[weight = 400_000]
+        pub fn store_valid_asset_ids(origin, // TODO: how to ensure only privileged users can call this?
+            valid_asset_ids: Vec<AssetId>,
+        ) -> DispatchResult
+        {
+            ensure_signed(origin)?;
+
+            <ValidAssetIds>::put(valid_asset_ids);
+            Ok(())
+        }
+
+        #[weight = 400_000]
+        pub fn store_account_tx(origin,
+            tx: PubAccountTx,
+        ) -> DispatchResult
+        {
+            let owner_acc = ensure_signed(origin)?;
+            let owner_id = Context::current_identity_or::<Identity<T>>(&owner_acc)?;
+
+            let account_vldtr = AccountValidator {};
+            let valid_asset_ids = Self::valid_asset_ids();
+            let valid_asset_ids = convert_asset_ids(valid_asset_ids);
+            let _ = account_vldtr.verify(&tx, &valid_asset_ids).map_err(|_| Error::<T>::InvalidAccountCreationProof)?;
+
+            let mut mercat_accounts = Self::mercat_accounts(owner_id);
+            mercat_accounts.push(tx.pub_account);
+            <MercatAccounts>::insert(&owner_id, mercat_accounts);
```

Comment:
```suggestion
            <MercatAccounts>::append(&owner_id, tx.pub_account);
```
(+ remove 2 lines before)